### PR TITLE
Fixes #2083: removed escaping of single quotes when saving

### LINKED ISSUE #2083: MapStore 2 escapes single quotes when saving
Try to change the title of a layer of the map as: 
```
CITTA' 
```
The map title will be converted into: 
```
"title": "CITTA\' "
```
That is not a valid JSON (because of the \').
Probably because of this: 
https://github.com/mathiasbynens/jsesc#quotes

### PR DESCRIPTION
**Pull request by mbarto**

### CODE REVIEW

File: web/client/api/GeoStoreDAO.js [hunk 1]
```diff
@@ -29,6 +29,10 @@ let parseUserGroups = (groupsObj) => {
     return groupsObj.User.groups.group.filter(obj => !!obj.id).map((obj) => _.pick(obj, ["id", "groupName", "description"]));
 };
 
+const encodeContent = function(content) {
+    return jsesc(content, {json: true, wrap: false, quotes: 'backtick'});
```
Comment (offtherailz) [inline web/client/api/GeoStoreDAO.js]: So the backtick will be escaped? Is there a way to disable quotes escaping totally ?

Comment (mbarto) [reply web/client/api/GeoStoreDAO.js]: Not with jsesc (unless we patch it). The options are:
 * forking and making a PR to the project
 * reimplementing the escaping code only for unicode chars, without using this lib

Comment (offtherailz) [reply web/client/api/GeoStoreDAO.js]: So if a user insert a backquote in a resource for instance, to break everything. We are planning to use markdown somewere for rich text, where backquote is part of the syntax.
I still didn't get why we have to use that lib. Did we do all the checks, as explained here? 
https://www.w3.org/International/questions/qa-changing-encoding

Comment (mbarto) [reply web/client/api/GeoStoreDAO.js]: Yes, it seems to be working only if we encode unicode chars with hex sequences. jsesc is doing this, but also does more and you cannot disable everything (like quote encoding). As I said, we can:
 * implement a flag in jsesc to only do unicode chars encoding
 * implement in MS2 a piece of code doing only that

Comment (offtherailz) [reply web/client/api/GeoStoreDAO.js]: This pull request moves the problem to a less used char. I think we should solve the problem at all.

**Pull request merged**